addon/components/-private/base-pop-menu.js

```diff
@@ -234,6 +258,29 @@ export default class BasePopMenuComponent extends Component {
     }
   };
 
+  /**
+   * Handler for closing the popper on target blur
+   */
+  _closePopoverBlurHandler = () => {
+    // Check if the popper has focusable elements. If it does, this means when the
+    // popper finishes rendering, it will move the focus to the first focusable item,
+    // and as a side effect trigger the blur event, but in this case we don't want
+    // the popper to close while we are still inside it. Once the user tabs off
+    // of the last focusable item in the popper, it will close anyway.
+    let focusableElements = this._getFocusableElementsInPopper();
+    if (focusableElements.length === 0) {
+      this._closePopoverHandler();
+    }
+  };
+
+  /**
+   * Closes the popover and focuses back on the trigger
+   */
+  _closePopoverAndFocusTrigger = () => {
```

Comment:
It looks like this function isn't actually an event handler, it's called within contexts that are already correctly bound. We should change it to a normal class method to be more performant.